compiler-rt/lib/rtsan/rtsan.cpp

```diff
@@ -49,7 +50,21 @@ static auto OnViolationAction(DiagnosticsInfo info) {
   return [info]() {
     IncrementTotalErrorCount();
 
-    PrintDiagnostics(info);
+    BufferedStackTrace stack;
+    stack.Unwind(info.pc, info.bp, nullptr,
+                 /*request_fast*/ true);
```

Comment:
It is unclear to me the pros/cons of requesting fast stack unwinding is, let me know if this is wrong.

My only gut feeling is "fast is good", but that's where my knowledge stops

Reply:
fast means frame pointer unwinding

Reply:
I see. 

Based on my understanding of the underlying implementation this is just a "pleasant request" to fast unwind, but it will be overridden when necessary (the platform does not support it?).

In this case, we are unwinding the stack, but it is not necessarily fatal, so the existing flags `fast_unwind_on_fatal` and `fast_unwind_on_malloc` seem inappropriate - should we make a new flag for this? or default to requesting "slow" unwinding?

This is definitely me at the end of my experience so I am happy to have any advice.

Reply:
e.g. hwasan also has a non-fatal mode, but still respects the `on_fatal` flag. so maybe it's fine to just use that.

Reply:
Took a look around and it seems like this is pretty common, despite whether or not the issue is fatal, the flag is used "on error".

I have moved to this approach, checking fast_unwind_on_fatal here.

Perhaps this flag should be renamed, but obviously that is a bigger discussion than this PR, and above my pay grade. Left a comment in the code describing what is going on

Reply:
i think the implication is "fast unwind on error report" vs malloc which is even done in non error cases.